UnitsNet/QuantityTypeConverter.cs

```diff
@@ -0,0 +1,205 @@
+using System.Globalization;
+using System.Runtime.CompilerServices;
+
+namespace UnitsNet
+{
+    /// <summary>
+    /// Is the base class for all attributes that are related to <see cref="UnitsNetTypeConverter{T}"/>
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+    public abstract class UnitAttributeBase : Attribute
+    {
+        /// <summary>
+        /// This property hold the UnitType information
+        /// </summary>
+        public Enum UnitType { get; set; }
+
+        /// <summary>
+        /// Constructor of this class
+        /// </summary>
+        /// <param name="unitType"></param>
+        public UnitAttributeBase(object unitType)
+        {
+            UnitType = unitType as Enum;
+        }
+    }
+
+    /// <summary>
+    /// This attribute defines the default Unit to use. It is only used when the string only consists of digits
+    /// </summary>
+    public class DefaultUnitAttribute : UnitAttributeBase
+    {
+        /// <summary>
+        /// Constructor for ConvertToUnit attribute
+        /// </summary>
+        /// <param name="unitType">The unit the quantity gets when the string parsing dose only consist of digits</param>
+        public DefaultUnitAttribute(object unitType) : base(unitType) { }
+    }
+
+    /// <summary>
+    /// This attribute defines the Unit the quantity is converted to after it has been parsed.
+    /// </summary>
+    public class ConvertToUnitAttribute : DefaultUnitAttribute
+    {
+        /// <summary>
+        /// Constructor for ConvertToUnit attribute
+        /// </summary>
+        /// <param name="unitType">The unit the quantity is converted to when parsing from string</param>
+        public ConvertToUnitAttribute(object unitType) : base(unitType) { }
+    }
+
+    /// <summary>
+    /// This attribute defines the unit the quantity has when converting to string
+    /// </summary>
+    public class DisplayAsUnitAttribute : DefaultUnitAttribute
+    {
+        /// <summary>
+        /// The formating used when the quantity is converted to string
+        /// </summary>
+        public string Format { get; set; }
+
+        /// <summary>
+        /// Constructor for DisplayAsUnit attribute
+        /// </summary>
+        /// <param name="unitType">The unit the quantity should be displayed in</param>
+        /// <param name="format">Formating string <see cref="IQuantity.ToString(System.IFormatProvider)"/></param>
+        public DisplayAsUnitAttribute(object unitType, string format = "") : base(unitType) { Format = format; }
+    }
+
+    /// <summary>
+    ///     Converts IQuantitys from string and to string.
+    ///     Implements the TypeConverter interface so that eg the PropertyGrid can read and write the properties implementing the IQuantity interface.
+    /// </summary>
+    /// <typeparam name="TQuantity">Quantity value type, such as <see cref="Length"/> or <see cref="Mass"/>.</typeparam>
```

Comment:
Should we have a short `<example></example>` here maybe? Or some links to documentation on how to use this?

Reply:
Nice example, wow!